accounts/views.py

```diff
@@ -41,4 +50,83 @@ def loginPage(request):
 
 def logoutUser(request):
     logout(request)
-    return redirect("/")
+    return redirect("/login")
+
+
+@login_required(login_url="/login")
+def profilePage(request):
+    correct_answers = (
+        Answer.objects.all()
+        .filter(user=request.user)
+        .aggregate(Sum("correct_count"))["correct_count__sum"]
+    )
+    incorrect_answers = (
+        Answer.objects.all()
+        .filter(user=request.user)
+        .aggregate(Sum("incorrect_count"))["incorrect_count__sum"]
+    )
+
+    result = (
+        Answer.objects.values("date")
+        .filter(user=request.user)
+        .order_by("date")
+        .annotate(correct=Sum("correct_count"))
+        .annotate(incorrect=Sum("incorrect_count"))
+    )
+    result = list(result)
+    data = {"dates": [], "correct": [], "incorrect": [], "all": [], "percentage": []}
+    correct_sum, incorrect_sum = 0, 0
+    for el in result:
+        correct_sum += el["correct"]
+        incorrect_sum += el["incorrect"]
+        data["dates"].append(str(el["date"]))
+        data["correct"].append(correct_sum)
+        data["incorrect"].append(incorrect_sum)
+        data["all"].append(correct_sum + incorrect_sum)
+        data["percentage"].append(correct_sum / (correct_sum + incorrect_sum) * 100)
+
+    answers = (
+        Answer.objects.values("flashcard_id")
+        .filter(Q(user=request.user.id) & (~Q(correct_count=0) | ~Q(incorrect_count=0)))
+        .annotate(correct=Sum("correct_count"))
+        .annotate(incorrect=Sum("incorrect_count"))
+    )
+    answers = list(answers)
+    print(answers)
+    answers.sort(key=lambda answer: answer["correct"] / (answer["correct"] + answer["incorrect"]))
+    if len(answers) > 3:
```

Comment:
dlaczego akurat >3?

Reply:
w sensie nie, ze sie dopierdalam, tylko chce wiedziec xD

Reply:
Uznałem, że jak ktoś ma jedną odpowiedź czy dwie to dziwnie mu je wyświetlać w "najgorszych odpowiedziach" xD ale w zasadzie można to usunąć albo jakąś zmienną tam wstawić